python/tests/nonreg/arcticdb/version_store/test_nonreg_specific.py

```diff
@@ -197,3 +198,100 @@ def test_batch_write_unicode_strings(lmdb_version_store):
     for _ in range(5):
         lib.batch_write(syms, data)
         lib.batch_append(syms, data)
+
+
+def test_update_with_empty_series_or_dataframe(lmdb_version_store):
+    # Non-regression test for https://github.com/man-group/ArcticDB/issues/892
+    lib = lmdb_version_store
+
+    symbol = "test_update_with_empty_dataframe"
+
+    lib.write(symbol, pd.DataFrame())
+
+    # This must not fail.
+    lib.update(symbol, pd.DataFrame())
+
+    symbol = "test_update_with_empty_series"
+
+    lib.write(symbol, pd.Series())
+
+    # This must not fail.
+    lib.update(symbol, pd.Series())
+
+
+def test_update_with_empty_dataframe_with_index(lmdb_version_store):
+    # Non-regression test for https://github.com/man-group/ArcticDB/issues/940
+    lib = lmdb_version_store
+
+    symbol = "test_update_with_empty_dataframe_with_index"
+
+    series = pd.Series(dtype="datetime64[ns]")
+    lib.write(symbol, series)
+
+    # This must not fail.
+    lib.read(symbol, as_of=0).data
```

Comment:
I'd like to make these tests more comprehensive, covering the entire matrix of:

- `write`, `append`, and `update` methods
- existing symbol is empty/has data
- dataframe being written/appended/updated is empty/has data
- static/dynamic schema

Reply:
Started with 2a8bb7a.